game/src/contrib/entities/AIFactory.java

```diff
@@ -0,0 +1,208 @@
+                        random.nextFloat(PATROUILLE_RADIUS_LOW, PATROUILLE_RADIUS_HIGH),
+                        random.nextInt(CHECKPOINTS_LOW, CHECKPOINTS_HIGH + 1),
+                        random.nextInt(PAUSE_TIME_LOW, PAUSE_TIME_HIGH + 1),
+                        modes[random.nextInt(0, modes.length)]);
+            }
+            case 1 -> {
+                return new RadiusWalk(
+                        random.nextFloat(RADIUS_WALK_LOW, RADIUS_WALK_HIGH),
+                        random.nextInt(BREAK_TIME_LOW, BREAK_TIME_HIGH + 1));
+            }
+            case 2 -> {
+                return new StaticRadiusWalk(
+                        random.nextFloat(STATIC_RADIUS_WALK_LOW, STATIC_RADIUS_WALK_HIGH),
+                        random.nextInt(STATIC_BREAK_TIME_LOW, STATIC_BREAK_TIME_HIGH + 1));
+            }
+            default -> throw new IndexOutOfBoundsException("This IdleAI does not exist");
+        }
+    }
+
+    /**
+     * Constructs a random TransitionAI with random parameters. Needs to be Updated whenever a new
+     * TransitionAI is added.
+     *
+     * @return the generated TransitionAI
+     */
+    public static Function<Entity, Boolean> generateRandomTransitionAI(Entity entity) {
+        Random random = new Random();
+        int index = random.nextInt(0, 4);
+
+        switch (index) {
+            case 0 -> {
+                return new RangeTransition(
+                        random.nextFloat(RANGE_TRANSITION_LOW, RANGE_TRANSITION_HIGH));
+            }
+            case 1 -> {
+                return new SelfDefendTransition();
+            }
+            case 2 -> {
+                ProtectOnApproach transition;
+                Optional<Entity> randomMonster = getRandomMonster();
+                if (getRandomMonster().isPresent()) {
+                    transition =
+                            new ProtectOnApproach(
+                                    random.nextFloat(PROTECT_RANGE_LOW, PROTECT_RANGE_HIGH),
+                                    randomMonster.get());
+                } else {
+                    transition =
+                            new ProtectOnApproach(
+                                    random.nextFloat(PROTECT_RANGE_LOW, PROTECT_RANGE_HIGH),
+                                    entity);
+                }
+                return transition;
+            }
+            case 3 -> {
+                ProtectOnAttack transition;
+                Optional<Entity> randomMonster = getRandomMonster();
+                if (getRandomMonster().isPresent()) {
+                    transition = new ProtectOnAttack(randomMonster.get());
+                } else {
+                    transition = new ProtectOnAttack(entity);
+                }
+                return transition;
+            }
+            default -> throw new IndexOutOfBoundsException("This TransitionAI does not exist");
+        }
+    }
+
+    /**
+     * Goes filters all Entities for Monsters and returns a random one.
```

Comment:
finde die doch etwas seltsam,
ich glaube "Sucht in den aktuell aktiven Entiäten nach Monstern und gibt zufällig eines davon zurück <p> Ein Monster ist eine Entität mit xyz Compoment <p> Methode wird genutzt um ein Parameter Monster für bestimme AI-Verhaltensweisen zu bekommen z.B XYZ"

Dann wird zumindest klar wofür das gut ist.